assets/src/edit-story/components/header/buttons.js

```diff
@@ -67,14 +67,14 @@ function PreviewButton() {
 
 function Publish() {
 	const {
-		state: { isSaving, postStatus },
-		actions: { savePost },
+		state: { isSaving, story: { status } },
+		actions: { saveStory },
 	} = useStory();
 
-	const text = ( postStatus !== 'publish' ) ? __( 'Publish' ) : __( 'Update' );
+	const text = ( status !== 'publish' ) ? __( 'Publish' ) : __( 'Update' );
 
 	return (
-		<Primary onClick={ savePost } isDisabled={ isSaving }>
+		<Primary onClick={ saveStory } isDisabled={ isSaving }>
```

Comment:
This PR removes `isSaving` state. Can we perhaps keep that to retain functionality (disable publish button while saving)?

Reply:
But where to put it? It doesn't belong inside the story reducer (as it's not part of the story, it's just UX meta). All the way back in the API would probably make the most sense, but then this publish button needs to know, which underlying API method (or even methods) are used for saving a story? It might go together with another variable remembering whether the story is dirty (changed since last save) in some state object on the story, e.g.:
```js
	const {
		state: {
			meta: { isSaving, isDirty },
			story: { status },
		},
		actions: { saveStory },
	} = useStory();
```

And then create a separate reducer for updating these two states when relevant?

Reply:
Good question. Your suggestion makes sense to me.

Perhaps there's gonna be more meta variables like this in the future too. #4020 comes to mind here.

Reply:
Done in bf5d6fb.

Reply:
(only the `isSaving` part though - `isDirty` is better dealt with through the history provider, I think)